### DISCUSSION TITLE
SPEC file questions

### DISCUSSION THREAD

**Discussion by Mogaba** [rpm-software-management/rpm#2331] [Q&A] [answered]
Hi,

I'm new to RPM package building and I have a couple of questions regarding SPEC file:

1. If I use the %defattr directive without specifying file attributes, like this:

   `%defattr(-,root,root,755)`

   does it mean file attributes will be the same as in %buildroot folder or attributes will be set at the installation time in accordance with umask?

2. If I use a wildcard in the %files section like this:

   `/usr/bin/*`

   does it mean I just added all files in `%buildroot/usr/bin/` to the RPM package or I made RPM think that all files in `/usr/bin/` belong to the package on the target system?

**Comment by pmatilai** [comment, marked as answer]
Oh, this has gone somehow entirely unnoticed, sorry about that.

1. %defattr(-,root,root,755) means that for regular files, the permission bits are collected from %buildroot, but directories will be set as 755.
2. The %files section always refers to the buildroot contents in modern rpm
